internal/service/advancedclustertpf/resource_schema.go

```diff
@@ -132,42 +148,66 @@ func ResourceSchema(ctx context.Context) schema.Schema {
-				Computed:            true,
+				Optional: true,
+				// Computed:            true,
+				// PlanModifiers: []planmodifier.String{
+				// 	stringplanmodifier.UseStateForUnknown(),
+				// },
 				MarkdownDescription: "Feature compatibility version of the cluster.",
 			},
 			"feature_compatibility_version_expiration_date": schema.StringAttribute{ // TODO: not in current resource, decide if keep
-				Computed:            true,
+				// Computed:            true,
+				Optional: true,
+				// PlanModifiers: []planmodifier.String{
+				// 	stringplanmodifier.UseStateForUnknown(),
+				// },
 				MarkdownDescription: "Feature compatibility version expiration date.",
 			},
 			"global_cluster_self_managed_sharding": schema.BoolAttribute{
-				Computed:            true,
-				Optional:            true,
+				Computed: true,
+				Optional: true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+				},
 				MarkdownDescription: "Set this field to configure the Sharding Management Mode when creating a new Global Cluster.\n\nWhen set to false, the management mode is set to Atlas-Managed Sharding. This mode fully manages the sharding of your Global Cluster and is built to provide a seamless deployment experience.\n\nWhen set to true, the management mode is set to Self-Managed Sharding. This mode leaves the management of shards in your hands and is built to provide an advanced and flexible deployment experience.\n\nThis setting cannot be changed once the cluster is deployed.",
 			},
 			"project_id": schema.StringAttribute{ // TODO: fail if trying to update
 				Required:            true,
 				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
 			},
 			"cluster_id": schema.StringAttribute{ // TODO: was generated as id
-				Computed:            true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies the cluster.",
 			},
-			"labels": schema.ListNestedAttribute{ // TODO: database_user is using SetNestedBlock, probably better to align
-				Computed:            true, // TODO: not sure if it should be computed
+			"labels": schema.SetNestedAttribute{ // TODO: database_user is using SetNestedBlock, probably better to align
+				Computed:            true, // TODO: must be computed since backend returns "[]" when it is not specified using "Default" to avoid plan changes
 				Optional:            true,
+				Default:             setdefault.StaticValue(types.SetValueMust(LabelsObjType, nil)),
```

Comment:
do we need this default? what's the difference with no having this line?

Reply:
```

Step 1/1 error: After applying this test step, the non-refresh plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        Terraform will perform the following actions:

          # mongodbatlas_advanced_cluster.test will be updated in-place
          ~ resource "mongodbatlas_advanced_cluster" "test" {
              ~ labels                               = [] -> (known after apply)
                name                                 = "test"
                # (21 unchanged attributes hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
        ```

Reply:
See also, comment two lines above

Reply:
sorry, can you confirm what comment?

Reply:
removed the old comment. Hopefully, more clear now 👍